pre/remap_restart/remap_questions.py

```diff
@@ -1,395 +1,448 @@
-      cmd = fvrst + fvcore
-      print(cmd +'\n')
-      p = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE)
-      (output, err) = p.communicate()
-      p_status = p.wait()
-      ss = output.decode().split()
-      x['input:shared:agrid'] = "C"+ss[0] # save for air parameter
-      lat = int(ss[0])
-      lon = int(ss[1])
-      if (lon != lat*6) :
-         sys.exit('This is not a cubed-sphere grid fvcore restart. Please contact SI team')
-      ymdh  = x.get('input:shared:yyyymmddhh')
-      ymdh_ = str(ss[3]) + str(ss[4])[0:2]
-      if (ymdh_ != ymdh) :
-         print("Warning: The date in fvcore is different from the date you input\n")
-      zoom = lat /90.0
-      zoom_ = str(int(zoom))
-      if zoom < 1 : zoom_ = '1'
-      if zoom > 8 : zoom_ = '8'
-   if x['input:shared:MERRA-2'] :
-      zoom_ = '2'
-   return zoom_
-
-def get_account():
-   cmd = 'id -gn'
-   p = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE)
-   (accounts, err) = p.communicate()
-   p_status = p.wait()
-   accounts = accounts.decode().split()
-   return accounts[0]
+    if ogrid == 'CS':
+       ogrid = x['output:shared:agrid']
+    x['output:shared:ogrid'] = ogrid
+  
+  return False
+
+def echo_level(x):
+  if x["output:air:nlevel"] != str(x.get("input:air:nlevel")) :
+      print("NOTE: Different # atm levels in input and new restarts.  Cannot remap agcm_import_rst (a.k.a. IAU) file.")
+      x['output:air:agcm_import_rst'] = False
+      return False
+  return True
+
+def echo_bcs(x,opt):
+  base      = x.get(opt+':shared:bc_base').split(': ')[-1]
+  bcv       = x.get(opt+':shared:bc_version')
+  agrid     = x.get(opt+':shared:agrid')
+  ogrid     = x.get(opt+':shared:ogrid')
+  model     = x.get(opt+':shared:omodel')
+  stretch   = x.get(opt+':shared:stretch')
+  x[opt+':shared:bc_base']  = base
+  land_dir  = get_landdir(base, bcv, agrid, ogrid, model, stretch)
+  if  not os.path.isdir(land_dir):
+     exit("cannot find grid subdirectory for agrid=" + agrid + " and ogrid=" + ogrid + " under " + base+'/'+bcv+'/land/')
+  print("\n Land BCs for " + opt + " restarts: " + land_dir )
+  return False
+
+def default_partition(x):
+   if x['slurm:qos'] == 'debug':
+      x['slurm:partition'] = 'compute'
+      return False
+   return True
+
+def validate_merra2_time(text):
+   if len(text) == 10 :
+      hh = text[8:]
+      if hh in ['03','09','15','21']:
+         return True
+      else:
+         return False 
+   else:
+     return False
 
 def ask_questions():
 
+   # See remap_utils.py for definitions of "choices", "message" strings, and "validate" lists
+   # that are used multiple times.
+
    questions = [
         {
             "type": "confirm",
             "name": "input:shared:MERRA-2",
-            "message": "Would you like to remap archived MERRA-2 restarts?",
+            "message": "Remap archived MERRA-2 restarts? (NCCS/Discover only; elsewhere, select 'N' and complete full config; requires nc4 restarts.)\n",
```

Comment:
@weiyuan-jiang, @biljanaorescanin : I assume the option to use archived M-2 restarts works on Discover only and clarified the question accordingly.  I may be wrong.  Scott and/or Matt can advise on whether this is something that should work at NAS, in which case more work is needed.  I doubt this is urgent, though, and it can be left for later.

Reply:
Most likely M2 restarts would have to be migrated to NAS. However, that is something which could be addressed later if there is a need to do so.